compile/inc/src/main/scala/sbt/inc/APIs.scala

```diff
@@ -38,6 +38,8 @@ trait APIs {
 
   def internal: Map[File, Source]
   def external: Map[String, Source]
+
+  def hasPackageObject(src: File): Boolean
```

Comment:
Adding a new method does not break binary compatibility. People who extend this trait will have to implement `hasPackageObject`, but classes that have already been compiled should not need to be recompiled.